### PR TITLE
Add style guidance for external links

### PR DESCRIPTION
**Pull request by ddbeck**
#### Summary

Add style guidance for external links

#### Motivation

I've been asked a few times recently about how to handle this or that external link. This is my attempt to put my approach to external links into writing.

#### Related issues

https://github.com/mdn/content/pull/13324

#### Metadata

- [x] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by ddbeck** [comment]
See https://deploy-preview-13899--mdn-contributor-docs.netlify.app/writing/style/external-links/ for a preview.

**Review by wbamberg — APPROVED** [review]
This looks great to me. One thing I consider when reviewing links, which isn't directly included here, is "likelihood of existing and being maintained in a few years". It might be worth adding under "Good external links" something like: links to resources that have a substantial track record of uptake and maintenance, like a library that has existed and been actively maintained for several years.

I think that's probably the biggest "problematic" category: obvious self-promotion is obvious, but what often happens is that people add links to new resources in good faith (and with the optimism of starting something new) and the resources are then not maintained (noone adopts it, the maintainer loses interest...) and we are stuck.

Also "Ephemeral or unmaintained" relates to blog posts, which are a tricky category. We do link to blog posts quite a lot, and they are problematic because while they are often very good, they are almost always unmaintained, and even good blog posts will be unhelpful a few years down the line. I wonder if we should discuss blog posts explicitly.

**Comment by hamishwillee** [comment]
I agree re blogs. That said, there are some specific blogs where the content ages well and they are maintained. It might be worth mentioning the list of these that are explicitly considered "ok". I'm thinking developer.google.com and friends.

Where would you position a wikipedia link in all this?

File: meta/content/en/writing/style/external-links.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+title: "External links"
+linkTitle: "External links"
+weight: 40
+description: >
+  How to choose whether to include an external link
+---
+
+External links are allowed on MDN,
+but they require extra care to make sure they’re useful to the MDN’s audience.
```
Comment (Elchi3) [inline meta/content/en/writing/style/external-links.md:10]:
```suggestion
but they require extra care to make sure they’re useful to MDN’s audience.
```

**Review by Elchi3 — APPROVED** [review]

**Comment by ddbeck** [comment]
@wbamberg You raise some good points here. One thing I tried (and perhaps failed) to capture is that I expect reviewers to weigh signs of a good link against signs of a bad link. A one-time announcement (that won't be updated) might still be the best link for the job, if there's no plausible (or even approximately as good) alternative. But if we have our choice between linking to, say, a blog post and a Wikipedia entry, I'd expect the Wikipedia entry to "win" on the merits of maintainability (all other things being equal). Every link will rot eventually, but my preference is prefer the links that rot slowly. I'm not quite sure the best way to get that across and I'm open to ideas here.

@hamishwillee Yeah, I suppose this might be a good place to document allowlist for known-good domains to link too. That said, I think assembling that list might provoke additional discussion (and have its own selection criteria process)—feels a bit out of scope for this PR, really.

**Comment by wbamberg** [comment]
Sure. I'm happy to merge this as it is and expand it later if we need to.

**Review by hamishwillee — APPROVED** [review]

**Review by Rumyra — APPROVED** [review]

**Comment by Rumyra** [comment]
I'm going to merge because it seems everyone has approved 👍

**Pull request merged**